### PR TITLE
feat(485): API lists installed Screwdriver packages and licenses

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Based on: https://github.com/screwdriver-cd/screwdriver/issues/485

Hitting https://api.screwdriver.cd/v4/versions will return something like this:

```json
{
    "versions": [
        "screwdriver-api@0.4.0",
        "screwdriver-artifact-bookend@1.1.1",
        "screwdriver-build-bookend@2.2.1",
        "screwdriver-config-parser@3.6.0",
        "screwdriver-data-schema@16.10.1",
        "screwdriver-datastore-base@3.0.1",
        "screwdriver-datastore-sequelize@2.1.1",
        "screwdriver-executor-base@5.2.2",
        "screwdriver-executor-docker@2.2.2",
        "screwdriver-executor-k8s@10.3.3",
        "screwdriver-models@22.2.4",
        "screwdriver-notifications-base@3.0.0",
        "screwdriver-notifications-email@1.1.2",
        "screwdriver-scm-base@2.7.0",
        "screwdriver-scm-bitbucket@2.8.1",
        "screwdriver-scm-github@4.8.2",
        "screwdriver-scm-gitlab@0.1.0",
        "screwdriver-template-validator@3.0.0"
    ],
    "licenses": [
        {
            "name": "@types/geojson",
            "repository": "https://wwwhub.com/DefinitelyTyped/DefinitelyTyped",
            "licenses": "MIT"
        },
        {
            "name": "JSONStream",
            "repository": "https://github.com/dominictarr/JSONStream",
            "licenses": "MIT*"
        },
        {
            "name": "abbrev",
            "repository": "https://github.com/isaacs/abbrev-js",
            "licenses": "ISC"
        },
        {
            "name": "accept",
            "repository": "https://github.com/hapijs/accept",
            "licenses": "BSD-3-Clause"
        },
        ...
    ]
}
```

### CODE REVIEW

File: plugins/versions.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Hapi interface for plugin to return package list
+ * @method register
+ * @param  {Hapi.Server}    server
+ * @param  {Object}         options
+ * @param  {Function} next
+ */
+exports.register = (server, options, next) => {
+    const SD_REGEX = /^screwdriver-/;
```
Comment (FenrirUnbound) [inline plugins/versions.js]: A comment example of what this regex is intended to match with would help with others debugging this in the future

File: plugins/versions.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            repository: dep.repository || 'UNKNOWN',
+            licenses: dep.licenses || 'UNKNOWN'
+        }));
+        const sdDeps = depArray.filter(dep => SD_REGEX.test(dep.name)).map(dep => dep.name);
+
+        server.route({
+            method: 'GET',
+            path: '/versions',
+            handler: (request, reply) => reply({
+                versions: sdDeps,
```
Comment (d2lam) [inline plugins/versions.js]: Wonder why it's `versions` and `licenses`? To me `sd-dependencies` and `other-dependences` make more sense

Comment (minz1027) [inline plugins/versions.js]: I think `other-dependences` will be too much, just look at how big it is in the build :(

File: plugins/versions.js (hunk 1 above)
Comment (stjohnjohnson) [inline plugins/versions.js]: Updated

File: plugins/versions.js (hunk 2 above)
Comment (stjohnjohnson) [inline plugins/versions.js]: versions are a list of Screwdriver packages

licenses are a list of all packages and their licenses

**Pull request merged**